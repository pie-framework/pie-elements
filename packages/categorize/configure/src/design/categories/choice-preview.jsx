import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { Choice } from '@pie-lib/pie-toolbox/drag';
import IconButton from '@material-ui/core/IconButton';
import RemoveCircleOutlineIcon from '@material-ui/icons/RemoveCircleOutline';
import { HtmlAndMath } from '@pie-lib/pie-toolbox/render-ui';

export class ChoicePreview extends React.Component {
  static propTypes = {
    alternateResponseIndex: PropTypes.number,
    category: PropTypes.object,
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    choice: PropTypes.object.isRequired,
    choiceIndex: PropTypes.number,
    onDelete: PropTypes.func.isRequired,
  };
  static defaultProps = {};

  delete = () => {
    const { onDelete, choice } = this.props;
    onDelete(choice);
  };

  render() {
    const { alternateResponseIndex, category, classes, className, choice, choiceIndex } = this.props;
    return (
      <div className={classNames(classes.choicePreview, className)}>
        {choice ? (
          <Choice alternateResponseIndex={alternateResponseIndex} category={category} choice={choice} choiceIndex={choiceIndex} onRemoveChoice={() => this.delete()}>
            <HtmlAndMath html={choice?.content} className={`${classes.breakWord}`} />
            <IconButton color="secondary" aria-label="delete" className={classes.delete} onClick={this.delete}>
              <RemoveCircleOutlineIcon />
            </IconButton>
          </Choice>
        ) : null}
      </div>
    );
  }
}
const styles = () => ({
  choicePreview: {
    position: 'relative',
  },
  delete: {
    position: 'absolute',
    right: 0,
    top: 0,
  },
  breakWord: {
    maxWidth: '90%',
    wordWrap: 'break-word',
  }
});
export default withStyles(styles)(ChoicePreview);
