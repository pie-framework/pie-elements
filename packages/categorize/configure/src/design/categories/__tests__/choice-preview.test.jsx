import { shallow } from 'enzyme';
import React from 'react';
import { ChoicePreview } from '../choice-preview';

describe('ChoicePreview', () => {
  let w;
  let onDelete = jest.fn();
  const wrapper = (extras) => {
    const defaults = {
      classes: {},
      className: 'className',
      choice: {
        content: 'content',
        id: '1',
      },
      onDelete,
    };
    const props = { ...defaults, ...extras };
    return shallow(<ChoicePreview {...props} />);
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
  describe('logic', () => {
    describe('delete', () => {
      it('calls onDelete', () => {
        w = wrapper();
        w.instance().delete();
        expect(onDelete).toBeCalledWith({ content: 'content', id: '1' });
      });
    });
  });
});
