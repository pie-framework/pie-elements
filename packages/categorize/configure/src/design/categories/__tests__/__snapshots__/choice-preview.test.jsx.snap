// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoicePreview snapshot renders 1`] = `
<div
  className="className"
>
  <Component>
    <HtmlAndMath
      html="content"
    />
    <WithStyles(IconButton)
      aria-label="delete"
      color="secondary"
      onClick={[Function]}
    >
      <pure(Delete) />
    </WithStyles(IconButton)>
  </Component>
</div>
`;
